GenericCollectionClass/GenericCollectionClass/SetWriterSpace/SetWriter.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace GenericCollectionClass.SetWriterSpace
+{
+    using System;
+    using System.IO;
+
+    public class SetWriter : IDisposable
+    {
+        private bool disposedValue;
+
+        public SetWriter(string filePath)
+        {
+            this.StreamWriter = new StreamWriter(filePath
+                ?? throw new ArgumentNullException($"{nameof(SetWriter)} cannot be instantiated because of a null filePath"));
+        }
+
+        private StreamWriter StreamWriter { get; }
+
+        public void WriteToStream(NumSet set)
+        {
+            this.StreamWriter.Write(set.ToString());
```

Comment:
```suggestion
            StreamWriter.Write(set.ToString());
```
You also may consider using `WriteLine` here.